frontend/src/components/Order/OrderButton/OrderButton.tsx

```diff
@@ -3,36 +3,92 @@ import { useFormContext } from 'react-hook-form';
 import { useNavigate } from 'react-router-dom';
 import { renderOrderSuccessToast } from '@/utils/toastContents.tsx';
 import { toast } from 'react-toastify';
+import useFetchProductData from '@/hooks/fetch/useFetchProductData.ts';
+import { useEffect } from 'react';
+import { getUserInfo } from '../../../../storage/userInfo.ts';
+import order from '../../../../api/order.ts';
+import { PATH } from '@/constants/path';
 
 interface Props {
   id: number;
   count: number;
-  receiveForm: {
-    submittedRef: React.MutableRefObject<{ count: number }[] | null>;
+  receiverForm: {
+    submittedRef: React.MutableRefObject<{ name: string; phone: string; count: number }[] | null>;
   };
 }
 
-export default function OrderButton({ id, ranking, count, receiverForm }: Props) {
+export default function OrderButton({ id, count, receiverForm }: Props) {
   const { handleSubmit } = useFormContext();
   const { submittedRef } = receiverForm;
   const navigate = useNavigate();
+  const { product, loading, error } = useFetchProductData(id);
 
-  const item = ranking.find((item) => Number(item.id) === id);
-  const price = item.price.sellingPrice * count;
+  useEffect(() => {
+    if (error) {
+      toast.error(error, {
+        autoClose: 3000,
+      });
+      navigate(`${PATH.HOME}`);
+    }
+  }, [error]);
 
-  const onSubmit = (data) => {
-    if (!submittedRef.current || submittedRef.current.length === 0) return;
+  if (!product) {
+    return <div>상품 정보가 없습니다.</div>
+  }
 
-    const { textMessage, senderName } = data;
+  const price = product.price * count;
 
-    toast(renderOrderSuccessToast(item.name, count, senderName, textMessage), {
-      type: 'success',
-      autoClose: 3000,
-      style: { width: '400px' },
-    });
+  const onSubmit = async (data) => {
+    const userInfo = getUserInfo();
+    const token = userInfo?.authToken;
 
-    navigate('/');
+    if (!token) {
+      toast.error('로그인이 필요합니다.');
+      navigate(`${PATH.LOGIN}`);
+      return;
+    }
+
+    if (!submittedRef.current || submittedRef.current.length === 0) {
+      toast.error('받는 사람 정보를 입력해주세요.');
+      return;
+    }
+
+    const { textMessage, senderName, messageCardId } = data;
+
+    const orderData = {
+      productId: product.id,
+      message: textMessage,
+      messageCardId: String(messageCardId),
+      ordererName: senderName,
+      receivers: submittedRef.current.map(({ name, phone, count }) => ({
+        name,
+        phoneNumber: phone,
+        quantity: Number(count),
+      })),
+    }
```

Comment:
OrderRequest 타입과 맞지 않고 있어요.
api 폴더의 타입도 내보내서 타입 안정성을 가져가보면 어떨까요?
```suggestion
    const orderData: OrderRequest = {
      productId: product.id,
      message: textMessage,
      messageCardId: String(messageCardId),
      orderName: senderName,
      receivers: submittedRef.current.map(({ name, phone, count }) => ({
        name,
        phoneNumber: phone,
        quantity: Number(count),
      })),
    }
```

Reply:
export로 타입을 지정하여 수정하였습니다.